hack/ocm-minikube-ramen.sh

```diff
@@ -0,0 +1,360 @@
+#!/bin/sh
+# shellcheck disable=1090,2046,2086
+set -x
+set -e
+trap 'set -- ${?}; trap - EXIT; eval ${exit_stack}; echo exit status: ${1}' EXIT
+trap 'trap - ABRT' ABRT
+trap 'trap - QUIT' QUIT
+trap 'trap - TERM' TERM
+trap 'trap - INT' INT
+trap 'trap - HUP' HUP
+exit_stack_push()
+{
+	{ set +x; } 2>/dev/null
+	exit_stack=${*}\;${exit_stack}
+	set -x
+}
+exit_stack_push unset -v exit_stack
+exit_stack_push unset -f exit_stack_push
+exit_stack_pop()
+{
+	IFS=\; read -r x exit_stack <<-a
+	${exit_stack}
+	a
+	eval ${x}
+	unset -v x
+}
+exit_stack_push unset -f exit_stack_pop
+rook_ceph_branch_checkout()
+{
+return
+	git --git-dir ${1}/.git --work-tree ${1} checkout main
+	exit_stack_push git --git-dir ${1}/.git --work-tree ${1} checkout -
+	#git --git-dir ${1}/.git fetch https://github.com/ShyamsundarR/ramen rook-setup:rook-setup
+	git --git-dir ${1}/.git fetch https://github.com/hatfieldbrian/ramen rook-setup:rook-setup
+	exit_stack_pop
+	git --git-dir ${1}/.git --work-tree ${1} checkout rook-setup
+	exit_stack_push git --git-dir ${1}/.git --work-tree ${1} checkout -
+}
+exit_stack_push unset -f rook_ceph_branch_checkout
+rook_ceph_branch_checkout_undo()
+{
+return
+	exit_stack_pop
+}
+exit_stack_push unset -f rook_ceph_branch_checkout_undo
+rook_ceph_deploy()
+{
+	rook_ceph_branch_checkout ${1}/..
+	PROFILE=${2} ${1}/minikube-rook-setup.sh create
+	PROFILE=${3} ${1}/minikube-rook-setup.sh create
+	PRIMARY_CLUSTER=${2} SECONDARY_CLUSTER=${3} ${1}/minikube-rook-mirror-setup.sh
+	PRIMARY_CLUSTER=${3} SECONDARY_CLUSTER=${2} ${1}/minikube-rook-mirror-setup.sh
+	PRIMARY_CLUSTER=${2} SECONDARY_CLUSTER=${3} ${1}/minikube-rook-mirror-test.sh
+	PRIMARY_CLUSTER=${3} SECONDARY_CLUSTER=${2} ${1}/minikube-rook-mirror-test.sh
+	rook_ceph_branch_checkout_undo
+}
+exit_stack_push unset -f rook_ceph_deploy
+rook_ceph_undeploy()
+{
+	rook_ceph_branch_checkout ${1}/..
+	PROFILE=${3} ${1}/minikube-rook-setup.sh delete
+	PROFILE=${2} ${1}/minikube-rook-setup.sh delete
+	rook_ceph_branch_checkout_undo
+}
+exit_stack_push unset -f rook_ceph_undeploy
+ramen_branch_name=pr47_pr58_rbac2360357
+exit_stack_push unset -v ramen_branch_name
+ramen_branch_build()
+{
+	set -- ${1} ${ramen_branch_name}
+	git --git-dir ${1}/.git --work-tree ${1} checkout main -b ${2}
+	exit_stack_push git --git-dir ${1}/.git --work-tree ${1} checkout -
+	git --git-dir ${1}/.git --work-tree ${1} pull --rebase https://github.com/ramendr/ramen main
```

Comment:
Ideally we should build current branch and not checkout/shift branches at all. I assume this is present to build the current working branch that can relatively pass e2e, but can we clean this up, and run the script from such a branch instead?

Reply:
I agree, no checkouts.  And that is correct reason.  Once this pull request and https://github.com/RamenDR/ramen/pull/58 or equivalent are merged into a branch, this script can be expected to work in the same branch it is run from, provided nothing new breaks it (e.g. missing changes to ocm-ramen-samples).  Issue https://github.com/RamenDR/ramen/issues/77 tracks the checkouts removal.